metaflow/plugins/argo/argo_workflows.py

```diff
@@ -1424,3 +1865,253 @@ def to_json(self):
+
+    def __str__(self):
+        return json.dumps(self.to_json(), indent=4)
+
+
+class Trigger(object):
+    # https://github.com/argoproj/argo-events/blob/master/api/sensor.md#argoproj.io/v1alpha1.Trigger
+
+    def __init__(self):
+        tree = lambda: defaultdict(tree)
+        self.payload = tree()
+
+    def template(self, trigger_template):
+        self.payload["template"] = trigger_template.to_json()
+        return self
+
+    def parameters(self, trigger_parameters):
+        if "parameters" not in self.payload:
+            self.payload["parameters"] = []
+        for trigger_parameter in trigger_parameters:
+            self.payload["parameters"].append(trigger_parameter.to_json())
+        return self
+
+    def policy(self, trigger_policy):
+        self.payload["policy"] = trigger_policy.to_json()
+        return self
+
+    def to_json(self):
+        return self.payload
+
+    def __str__(self):
+        return json.dumps(self.to_json(), indent=4)
+
+
+class TriggerTemplate(object):
+    # https://github.com/argoproj/argo-events/blob/master/api/sensor.md#argoproj.io/v1alpha1.TriggerTemplate
+
+    def __init__(self, name):
+        tree = lambda: defaultdict(tree)
+        self.payload = tree()
+        self.payload["name"] = name
+
+    def argo_workflow_trigger(self, argo_workflow_trigger):
+        self.payload["argoWorkflow"] = argo_workflow_trigger.to_json()
+        return self
+
+    def to_json(self):
+        return self.payload
+
+    def __str__(self):
+        return json.dumps(self.payload, indent=4)
+
+
+class ArgoWorkflowTrigger(object):
+    # https://github.com/argoproj/argo-events/blob/master/api/sensor.md#argoproj.io/v1alpha1.ArgoWorkflowTrigger
+
+    def __init__(self):
+        tree = lambda: defaultdict(tree)
+        self.payload = tree()
+        self.payload["operation"] = "submit"
+        self.payload["group"] = "argoproj.io"
+        self.payload["version"] = "v1alpha1"
+        self.payload["resource"] = "workflows"
+
+    def source(self, source):
+        self.payload["source"] = source
+        return self
+
+    def parameters(self, trigger_parameters):
+        if "parameters" not in self.payload:
+            self.payload["parameters"] = []
+        for trigger_parameter in trigger_parameters:
+            self.payload["parameters"].append(trigger_parameter.to_json())
+        return self
+
+    def to_json(self):
+        return self.payload
+
+    def __str__(self):
+        return json.dumps(self.payload, indent=4)
+
+
+class TriggerParameter(object):
+    # https://github.com/argoproj/argo-events/blob/master/api/sensor.md#argoproj.io/v1alpha1.TriggerParameter
+
+    def __init__(self):
+        tree = lambda: defaultdict(tree)
+        self.payload = tree()
+
+    def src(self, dependency_name, data_key, value):
+        self.payload["src"] = {
+            "dependencyName": dependency_name,
+            "dataKey": data_key,
+            "value": value,
+            "useRawData": True,
```

Comment:
In my tests... this needed to be set to `False` in order for the payload to render correctly as an env var in the triggered workflow.

Not sure what we are trying to test - if the final sink for this data is always an env var, we should always stringify it (i.e. not raw).

cc @savingoyal

Reply:
To add - we can also leave this unset (let it default to False).